### PR TITLE
Make repeat interval customizable

### PR DESCRIPTION
**Pull request by utkuturunc**

### CODE REVIEW

File: server-payment/src/services/DonationManagerService.ts [hunk 1]
```diff
@@ -30,22 +30,46 @@ export class DonationManagerService implements IDonationManagerService {
   private getPackageForDonationCreator = async (donationCreator: IDonationCreator) => {
     const pkg = await this.packageService.getById(donationCreator.packageId);
     if (!pkg) throw new Error("Package does not exist, cannot create donation.");
+    console.log({ donationCreator });
```
Comment (helloanil) [inline server-payment/src/services/DonationManagerService.ts]: console.log :(

File: frontend-payment/src/support-page/form/form-components/package/PackageSelect.js [hunk 2]
```diff
@@ -33,9 +33,11 @@ const PackageSelect = props => {
             size="large"
             loading={loading}
             disabled={loading}
-            onSelect={value =>
-              selectPackage(packages.find(p => p.id === value))
-            }
+            onSelect={value => {
+              const pkg = packages.find(p => p.id === value);
+              selectPackage(pkg);
+              props.onPackageSelect && props.onPackageSelect(pkg);
```
Comment (helloanil) [inline frontend-payment/src/support-page/form/form-components/package/PackageSelect.js]: Let's move this to another function above called `onSelect`

File: frontend-payment/src/support-page/form/form-components/package/PackageCustomize.js [hunk 3]
```diff
@@ -3,58 +3,103 @@ import { Form, Select, InputNumber } from "antd";
...
+        {getFieldDecorator(currencyField)(
           <Select
             size="large"
-            defaultValue="USD"
+            disabled={
+              !selectedPackage ||
+              (selectedPackage &&
+                !selectedPackage.customizationConfig
+                  .allowPriceCurrencyCustomization)
+            }
```
Comment (helloanil) [inline frontend-payment/src/support-page/form/form-components/package/PackageCustomize.js]: Let's move this to another function above called `checkSelectIsDisabled`. We're also duplicating this so another function would be even more useful

File: server-payment/src/services/DonationManagerService.ts (hunk 1 above)
Comment (utkuturunc) [reply server-payment/src/services/DonationManagerService.ts]: done

File: frontend-payment/src/support-page/form/form-components/package/PackageSelect.js (hunk 2 above)
Comment (utkuturunc) [reply frontend-payment/src/support-page/form/form-components/package/PackageSelect.js]: done

File: frontend-payment/src/support-page/form/form-components/package/PackageCustomize.js (hunk 3 above)
Comment (utkuturunc) [reply frontend-payment/src/support-page/form/form-components/package/PackageCustomize.js]: done

**Pull request merged**